app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -212,54 +212,75 @@ class AgendaRepositoryImpl(
-            it.modifiedType == ModifiedType.DELETE
-        }.map {
-            it.id
-        }
-        val taskDeleteIds = db.getTaskDao().getModifiedTasks().first().filter {
-            it.modifiedType == ModifiedType.DELETE
-        }.map {
-            it.id
-        }
-        val eventDeleteIds = db.getEventDao().getModifiedEvents().first().filter {
-            it.modifiedType == ModifiedType.DELETE
-        }.map {
-            it.id
-        }
-
-        if (reminderDeleteIds.isNotEmpty() ||
-            taskDeleteIds.isNotEmpty() ||
-            eventDeleteIds.isNotEmpty()
-        ) {
-            val syncAgendaRequest = SyncAgendaRequest(
-                eventDeleteIds,
-                taskDeleteIds,
-                reminderDeleteIds
-            )
-            val results = getResourceResult { agendaApi.syncAgendaItems(syncAgendaRequest) }
-            when (results) {
-                is Resource.Error -> {
-                    Log.e(
-                        "SyncAgendaItem Error",
-                        results.message?.asString(appContext) ?: "Unknown Error"
-                    )
+        val coroutine = CoroutineScope(Dispatchers.IO + SupervisorJob())
+        return coroutine.async {
+            val reminderDeleteIdsAsync = async {
+                db.getReminderDao().getModifiedReminders().first().filter {
+                    it.modifiedType == ModifiedType.DELETE
+                }.map {
+                    it.id
                 }
-                is Resource.Success -> {
-                    reminderDeleteIds.forEach {
-                        db.getReminderDao().deleteModifiedReminderById(it)
-                    }
-                    taskDeleteIds.forEach {
-                        db.getTaskDao().deleteModifiedTaskById(it)
+            }
+            val taskDeleteIdsAsync = async {
+                db.getTaskDao().getModifiedTasks().first().filter {
+                    it.modifiedType == ModifiedType.DELETE
+                }.map {
+                    it.id
+                }
+            }
+            val eventDeleteIdsAsync = async {
+                db.getEventDao().getModifiedEvents().first().filter {
+                    it.modifiedType == ModifiedType.DELETE
+                }.map {
+                    it.id
+                }
+            }
+            val eventDeleteIds = eventDeleteIdsAsync.await()
+            val taskDeleteIds = taskDeleteIdsAsync.await()
+            val reminderDeleteIds = reminderDeleteIdsAsync.await()
+
+            if (reminderDeleteIds.isNotEmpty() ||
+                taskDeleteIds.isNotEmpty() ||
+                eventDeleteIds.isNotEmpty()
+            ) {
+                val syncAgendaRequest = SyncAgendaRequest(
+                    eventDeleteIds,
+                    taskDeleteIds,
+                    reminderDeleteIds
+                )
+                val results = getResourceResult { agendaApi.syncAgendaItems(syncAgendaRequest) }
```

Comment:
Aren't you just syncing the locally deleted agenda items here? The function is called `syncModifiedAgendaItems`